### PR TITLE
bsunanda:Run2-gem23 Try to remove overlap of GE11 detector in 2017/18 scenario

### PR DESCRIPTION
**Pull request by bsunanda**

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna Could you do the overlap test for GEM detector (also entire ME) now?

**Comment by ianna** [comment]
@bsunanda - sure, I will

File: Geometry/CMSCommonData/python/cmsExtendedGeometry2017Plan1XML_cfi.py [hunk 1]
```diff
@@ -230,7 +230,7 @@
         'Geometry/MuonCommonData/data/Run2/mb3.xml',
         'Geometry/MuonCommonData/data/Run2/mb4.xml',
         'Geometry/MuonCommonData/data/design/muonYoke.xml',
-        'Geometry/MuonCommonData/data/v2/mf.xml',
+        'Geometry/MuonCommonData/data/v6/mf.xml',
```
Comment (ianna) [inline Geometry/CMSCommonData/python/cmsExtendedGeometry2017Plan1XML_cfi.py]: @bsunanda - this removes GE11 extrusion, but introduces an overlap of ME1RingP with SupportTubeForHE:
```
 = Overlap ov02148: mf:MEP/mf:ME1RingP_1 overlapping mf:MEP/mfshield:SupportTubeForHE_1 ovlp=3.3
 - first volume: mf:ME1RingP at position:
matrix  - tr=0  rot=0  refl=0  scl=0
  1.000000    0.000000    0.000000    Tx =   0.000000
  0.000000    1.000000    0.000000    Ty =   0.000000
  0.000000    0.000000    1.000000    Tz =   0.000000
*** Shape mf:ME1Ring: TGeoPcon ***
    Nz    = 4
    phi1  =     0.00000
    dphi  =   360.00000
     plane 0: z=  564.10000 Rmin=  115.00000 Rmax=  280.00000
     plane 1: z=  567.40000 Rmin=  115.00000 Rmax=  280.00000
     plane 2: z=  567.40000 Rmin=   85.00000 Rmax=  280.00000
     plane 3: z=  634.50000 Rmin=   85.00000 Rmax=  280.00000
 Bounding box:
*** Shape mf:ME1Ring: TGeoBBox ***
    dX =   280.00000
    dY =   280.00000
    dZ =    35.20000
    origin: x=    0.00000 y=    0.00000 z=  599.30000
 - second volume: mfshield:SupportTubeForHE at position:
matrix  - tr=0  rot=0  refl=0  scl=0
  1.000000    0.000000    0.000000    Tx =   0.000000
  0.000000    1.000000    0.000000    Ty =   0.000000
  0.000000    0.000000    1.000000    Tz =   0.000000
*** Shape mfshield:SupportTubeForHE: TGeoPcon ***
    Nz    = 7
    phi1  =     0.00000
    dphi  =   360.00000
     plane 0: z=  554.10000 Rmin=   67.50000 Rmax=  251.00000
     plane 1: z=  564.10000 Rmin=   67.50000 Rmax=  251.00000
     plane 2: z=  564.10000 Rmin=   67.50000 Rmax=  130.00000
     plane 3: z=  567.40000 Rmin=   67.50000 Rmax=  130.00000
     plane 4: z=  567.40000 Rmin=   67.50000 Rmax=   85.00000
     plane 5: z=  659.00000 Rmin=   67.50000 Rmax=   85.00000
     plane 6: z=  683.50000 Rmin=   70.10000 Rmax=   85.00000
 Bounding box:
*** Shape mfshield:SupportTubeForHE: TGeoBBox ***
    dX =   251.00000
    dY =   251.00000
    dZ =    64.70000
    origin: x=    0.00000 y=    0.00000 z=  618.80000


```

**Comment by ianna** [comment]
-1

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@ianna Updated the shield part

File: Geometry/MuonCommonData/data/v6/mfshield.xml [hunk 2]
```diff
@@ -0,0 +1,632 @@
+<?xml version="1.0"?>
```
Comment (ianna) [inline Geometry/MuonCommonData/data/v6/mfshield.xml]: The difference between previous version and this one is more then adjusting an Air volume:
```
6,12c6,12
<    <ZSection z="5541*mm" rMin="675*mm" rMax="2510*mm"/>
<    <ZSection z="5641*mm" rMin="675*mm" rMax="2510*mm"/>
<    <ZSection z="5641*mm" rMin="675*mm" rMax="1150*mm"/>
<    <ZSection z="5674*mm" rMin="675*mm" rMax="1150*mm"/>
<    <ZSection z="5674*mm" rMin="675*mm" rMax="850*mm"/>
<    <ZSection z="6590*mm" rMin="675*mm" rMax="850*mm"/>
<    <ZSection z="6835*mm" rMin="701*mm" rMax="850*mm"/>
---
>    <ZSection z="5.541*m  " rMin="67.5*cm " rMax="2.51*m  "/>
>    <ZSection z="5.641*m  " rMin="67.5*cm " rMax="2.51*m  "/>
>    <ZSection z="5.641*m  " rMin="67.5*cm " rMax="1.3*m  "/>
>    <ZSection z="5.674*m  " rMin="67.5*cm " rMax="1.3*m  "/>
>    <ZSection z="5.674*m  " rMin="67.5*cm " rMax="85*cm "/>
>    <ZSection z="6590*mm" rMin="67.5*cm " rMax="85*cm "/>
>    <ZSection z="6.835*m  " rMin="701*mm" rMax="85*cm "/>
14,27c14,25
<   <Tubs name="ShieldingBoronPoly_0_ME1"    rMin="1150*mm" rMax="1227*mm" dz="16*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_0_ME1" rMin="1227*mm" rMax="1240*mm" dz="42.5*mm"  startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_1_ME1"    rMin="936*mm"  rMax="1227*mm" dz="20*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_1_ME1" rMin="965*mm"  rMax="1227*mm" dz="6.5*mm"  startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_2_ME1"    rMin="850*mm"  rMax="880*mm"  dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_2_ME1" rMin="880*mm"  rMax="893*mm"  dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_3_ME1"    rMin="893*mm"  rMax="923*mm"  dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_3_ME1" rMin="923*mm"  rMax="936*mm"  dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_4_ME1"    rMin="850*mm"  rMax="880*mm"  dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_4_ME1" rMin="880*mm"  rMax="893*mm"  dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_5_ME1"    rMin="893*mm " rMax="923*mm"  dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_5_ME1" rMin="923*mm"  rMax="936*mm"  dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingAntinomyLead_6_ME1" rMin="936*mm"  rMax="2330*mm" dz="5*mm " startPhi="0*deg" deltaPhi="360*deg"/>
<   <Tubs name="ShieldingBoronPoly_6_ME1"    rMin="936*mm"  rMax="2330*mm" dz="12*mm" startPhi="0*deg" deltaPhi="360*deg"/>
---
>   <Tubs name="ShieldingBoronPoly_1_ME1" rMin="965*mm" rMax="1240*mm" dz="2*cm " startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_1_ME1" rMin="965*mm" rMax="1240*mm" dz="6.49995*mm " startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingBoronPoly_2_ME1" rMin="85.1*cm " rMax="88.1*cm " dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_2_ME1" rMin="88.2*cm " rMax="89.5*cm " dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingBoronPoly_3_ME1" rMin="89.6*cm " rMax="941*mm" dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_3_ME1" rMin="942*mm" rMax="955*mm" dz="148.5*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingBoronPoly_4_ME1" rMin="85.1*cm " rMax="88.1*cm " dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_4_ME1" rMin="88.2*cm " rMax="89.5*cm " dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingBoronPoly_5_ME1" rMin="89.6*cm " rMax="931*mm" dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_5_ME1" rMin="932*mm" rMax="945*mm" dz="136*mm" startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingAntinomyLead_6_ME1" rMin="945.1*mm" rMax="2330*mm" dz="5*mm " startPhi="0*deg" deltaPhi="360*deg"/>
>   <Tubs name="ShieldingBoronPoly_6_ME1" rMin="945.1*mm" rMax="2330*mm" dz="11.9*mm" startPhi="0*deg" deltaPhi="360*deg"/>
57,64d54
<   <LogicalPart name="ShieldingBoronPoly_0_ME1" category="unspecified">
<    <rSolid name="ShieldingBoronPoly_0_ME1"/>
<    <rMaterial name="materials:Boron Polyethyl."/>
<   </LogicalPart>
<   <LogicalPart name="ShieldingAntinomyLead_0_ME1" category="unspecified">
<    <rSolid name="ShieldingAntinomyLead_0_ME1"/>
<    <rMaterial name="materials:AntiLead"/>
<   </LogicalPart>
214,225d203
<    <rChild name="mfshield:ShieldingBoronPoly_0_ME1"/>
<    <rRotation name="rotations:000D"/>
<    <Translation x="0*fm " y="0*fm " z="5658*mm"/>
<   </PosPart>
<   <PosPart copyNumber="1">
<    <rParent name="mf:ME1RingP"/>
<    <rChild name="mfshield:ShieldingAntinomyLead_0_ME1"/>
<    <rRotation name="rotations:000D"/>
<    <Translation x="0*fm " y="0*fm " z="5684.5*mm"/>
<   </PosPart>
<   <PosPart copyNumber="1">
<    <rParent name="mf:ME1RingP"/>
228c206
<    <Translation x="0*fm " y="0*fm " z="5694*mm"/>
---
>    <Translation x="0*fm " y="0*fm " z="5698*mm"/>
234c212
<    <Translation x="0*fm " y="0*fm " z="5720.5*mm"/>
---
>    <Translation x="0*fm " y="0*fm " z="5724.5*mm"/>
298,309d275
<    <rChild name="mfshield:ShieldingBoronPoly_0_ME1"/>
<    <rRotation name="rotations:000D"/>
<    <Translation x="0*fm " y="0*fm " z="5658*mm"/>
<   </PosPart>
<   <PosPart copyNumber="1">
<    <rParent name="mf:ME1RingN"/>
<    <rChild name="mfshield:ShieldingAntinomyLead_0_ME1"/>
<    <rRotation name="rotations:000D"/>
<    <Translation x="0*fm " y="0*fm " z="5684.5*mm"/>
<   </PosPart>
<   <PosPart copyNumber="1">
<    <rParent name="mf:ME1RingN"/>
312c278
<    <Translation x="0*fm " y="0*fm " z="5694*mm"/>
---
>    <Translation x="0*fm " y="0*fm " z="5698*mm"/>
318c284
<    <Translation x="0*fm " y="0*fm " z="5720.5*mm"/>
---
>    <Translation x="0*fm " y="0*fm " z="5724.5*mm"/>
```

Comment (ianna) [inline Geometry/MuonCommonData/data/v6/mfshield.xml]: The shield before and after modification:
screen shot 2017-03-09 at 08 42 19
screen shot 2017-03-09 at 08 42 59
Note, there are changes in both Boron Polyethil. and AntiLead shapes - an outer ring of the latter is added:
screen shot 2017-03-09 at 08 46 15

**Comment by ianna** [comment]
@bsunanda - does it correspond to actual shielding?

@civanch and @mdhildreth - the overlap with GE11 is fixed, but there are small changes in support structure. Please, let me know if it's still ok to include it in a GT update. Thanks.

**Comment by bsunanda** [comment]
@ianna I checked the support structure with the Integration Engineers while writing the GEM geometry and this came from the discussion.

**Comment by ianna** [comment]
@bsunanda - Please, close this PR. I've included this PR in #17844 and integrated it in a new directory structure as suggested by @davidlange6:

 _Subsystem/Package/data/fileName/Year/Version/fileName.xml_ 

When it is integrated in IB, I'd suggest you to restructure GEM, Hcal and HGCal data. The advantage is that the Spec files such as XYZnumbering.xml can co-exist in a more transparent structure having the same name but different versions.

**Comment by civanch** [comment]
@bsunanda , @ianna , what I see with Geant4 tests in this PR:
1) Overlaps for (GEMDisk11N and ME1RingP), (GEMDisk11N and ME1RingN) are fixed
2) surprisingly, CornaR_CSC_for_TotemT1_Plane_1_5_1 also fixed
3) new problem appears which I never saw before: 
          Overlap with volume already placed !
          Overlap is detected for volume ME41Space
          apparently fully encapsulating volume ME41Space
instead of old warning:
          Overlap is detected for volume ME41Space
          with ME41Space volume's
          local point (-1479.51,1750.05,-122.7), overlapping by at least: 2.6 mm 

Does it means that there are several copy-numbers for   ME41Space? Also note that the same report is repeated for ME31Space, ME32Space, ME42Space, YE1p_b, YE2p_a, YE3p_b.

**Comment by ianna** [comment]
@civanch - I do not see any problem with ME41Space (copy 1 in red and copy 2 in blue):
screen shot 2017-03-09 at 12 21 37

**Comment by ianna** [comment]
However, there is an extrusion:
```
 = Overlap ov02170: csc:ME41AlumFrame extruded by: csc:ME41AlumFrame/csc:ME41FR4Body_1 ovlp=0.6096
 - first volume: csc:ME41AlumFrame at position:
matrix Identity - tr=0  rot=0  refl=0  scl=0
  1.000000    0.000000    0.000000    Tx =   0.000000
  0.000000    1.000000    0.000000    Ty =   0.000000
  0.000000    0.000000    1.000000    Tz =   0.000000
*** TGeoCompositeShape : csc:ME41AlumFrame = 
 Bounding box:
*** Shape csc:ME41AlumFrame: TGeoBBox ***
    dX =    76.54500
    dY =     9.02335
    dZ =    83.35000
    origin: x=    0.00000 y=    0.00000 z=    0.00000
 - second volume: csc:ME41FR4Body at position:
matrix  - tr=0  rot=0  refl=0  scl=0
  1.000000    0.000000    0.000000    Tx =   0.000000
  0.000000    1.000000    0.000000    Ty =   0.000000
  0.000000    0.000000    1.000000    Tz =   0.000000
*** Shape csc:ME41FR4Body: TGeoArb8 ***
    point #0 : x=  -44.98500 y=   -8.41375 z=  -83.35000
    point #1 : x=  -44.98500 y=    8.41375 z=  -83.35000
    point #2 : x=   44.98500 y=    8.41375 z=  -83.35000
    point #3 : x=   44.98500 y=   -8.41375 z=  -83.35000
    point #4 : x=  -76.54500 y=   -8.41375 z=   83.35000
    point #5 : x=  -76.54500 y=    8.41375 z=   83.35000
    point #6 : x=   76.54500 y=    8.41375 z=   83.35000
    point #7 : x=   76.54500 y=   -8.41375 z=   83.35000
 Bounding box:
*** Shape csc:ME41FR4Body: TGeoBBox ***
    dX =    76.54500
    dY =     8.41375
    dZ =    83.35000
    origin: x=    0.00000 y=    0.00000 z=    0.00000

```
screen shot 2017-03-09 at 12 59 41

**Comment by civanch** [comment]
@bsunanda , @ianna , the overlap:
Overlap with mother volume !
          Overlap is detected for volume ME41FR4Body
          with its mother volume ME41AlumFrame
          at mother local point (245.214,-84.1375,-139.56), overlapping by at least: 6.096 mm 

exist for long time and is unchanged.

I have only one explanation for the  warning:
          Overlap is detected for volume ME41Space
          apparently fully encapsulating volume ME41Space

as this volume is instantiated twice. Nothing else come to my mind.

**Comment by bsunanda** [comment]
@ianna @civanch These volumes (ME41....) are in the definition of CSC. I shall take a look at this.

**Comment by bsunanda** [comment]
@ianna @civanch @davidlange6 According to Yana's suggestion I am closing this PR. Hope the suggested changes here are included in other PR which gets integrated. Yana - can you explain a bit about the convention suggested by David. I understood Subsystem (= Geometry) / Package (= xxxCommonData). But why you bring in fileName in between - I can have Year and Version to play with.

**Comment by davidlange6** [comment]
> On Mar 9, 2017, at 3:45 PM, bsunanda <notifications@github.com> wrote:
> 
> @ianna @civanch @davidlange6 According to Yana's suggestion I am closing this PR. Hope the suggested changes here are included in other PR which gets integrated. Yana - can you explain a bit about the convention suggested by David. I understood Subsystem (= Geometry) / Package (= xxxCommonData). But why you bring in fileName in between - I can have Year and Version to play with.

because its the files that get versioned not the directories.

**Comment by civanch** [comment]
@bsunanda  @ianna , concerning this ME41 I do not understand why there is sub-division on 9 sub-volumes. I naively think that there always should be sub-division to 8.

**Comment by bsunanda** [comment]
I am closing this PR in view of Ianna's comments

**Comment by ianna** [comment]
@civanch - 9 subdivisions are correct (the 1st is highlighted in red, the 9th in blue):
screen shot 2017-03-10 at 09 50 15

**Comment by ianna** [comment]
@civanch - ME41Space copy 1 shares a surface with copy 2, but they do not overlap:

screen shot 2017-03-10 at 10 05 00
screen shot 2017-03-10 at 10 05 29

**Pull request closed without merging**